hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java

```diff
@@ -190,10 +190,10 @@
 
   // thread executor shared by all Table instances created
   // by this connection
-  private volatile ExecutorService batchPool = null;
+  private volatile ThreadPoolExecutor batchPool = null;
```

Comment:
Why change this to ThreadPoolExecutor? Because MetricsConnection need this?

Reply:
Yes, the MetricsConnection need to call the thread count related methods in ThreadPoolExecutor.